src/kitovu/sync/kitovu_filecache.py

```diff
@@ -0,0 +1,68 @@
+"""FileCache keeps track of the state of the files, remotely and locally. It exists to determine if the local file
+has been changed between two synchronisation processes and allows for a conflict handling accordingly."""
+
+from enum import Enum
+import json
+import pathlib
+import typing
+
+from kitovu.sync import syncplugin
+from kitovu import utils
+
+# sync-process:
+# every time we sync something, the FileCache gets read, upon which kitovu decides what needs to be downloaded.
+# The files get downloaded, for each file, the FileCache gets updated for the files that were synchronised.
+
+
+class Filestate(Enum):
+    """Used to discern in which places files have changed"""
+    LOCAL = 1
+    REMOTE = 2
+    BOTH = 3
+    NONE = 4
+
+
+class File:
+    def __init__(self, local_digest_at_synctime: str,
+                 relative_path_with_filename: pathlib.PurePath,
+                 plugin: syncplugin.AbstractSyncPlugin) -> None:
+        self._cached_digest = local_digest_at_synctime
+        self._relative_path_with_filename: pathlib.PurePath
+        self._plugin = plugin
+        self._remote_digest = ""
+        self._local_digest = ""
+
+
+class FileCache:
+    def __init__(self, filename: str, path_to_json: pathlib.Path):
+        self._FILECACHE = filename  # filecache.json
+        self._path = path_to_json
+
+    def write_filecache_to_jsonfile(self, file: typing.List[File]):
```

Comment:
This might work, but is not what I meant exactly earlier - basically, I'd say the `FileCache` manages itself (and the caller can add individual files), and then you can tell it "save yourself" and "restore yourself".